R/version_check.R

```diff
@@ -0,0 +1,50 @@
+# Utilities for detecting new version
+#
+# Author: kimyen
+###############################################################################
+
+.RAN <- "https://sage-bionetworks.github.io/ran"
+.PACKAGE_NAME = "synapser"
+
+# convert the full version to a precision
+# precision can be one of the following values:
+.MAJOR <- 1
+.MINOR <- 2
+.PATCH <- 3
+.simplifyVersion <- function(version, precision) {
+  if (!match(precision, c(.MAJOR, .MINOR, .PATCH))) {
+    throw("Invalid precision: ", precision)
```

Comment:
If I'm not mistaken, `throw()` is from the package R.methodsS3 so we'd need to take it on as a dependency and write this as `R.methodsS3::throw()` (or just use `stop()`)

Reply:
oh, I didn't know that. Thank you!